specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/services.json

```diff
@@ -0,0 +1,513 @@
+      "get": {
+        "x-ms-examples": {
+          "CosmosDBServicesList": {
+            "$ref": "./examples/CosmosDBServicesList.json"
+          }
+        },
+        "operationId": "Service_List",
+        "description": "Gets the status of service.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The ServiceResource.",
+            "schema": {
+              "$ref": "#/definitions/ServiceResourceListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": ".../../cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/services/{serviceName}": {
+      "put": {
+        "tags": [
+          "DataTransfer",
+          "SqlDedicatedGateway"
+        ],
+        "x-ms-examples": {
+          "DataTransferServiceCreate": {
+            "$ref": "./examples/CosmosDBDataTransferServiceCreate.json"
+          },
+          "SqlDedicatedGatewayServiceCreate": {
+            "$ref": "./examples/CosmosDBSqlDedicatedGatewayServiceCreate.json"
+          }
+        },
+        "operationId": "Service_Create",
+        "description": "Creates a service.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/serviceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "servicesCreateUpdateParameter",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ServicesCreateUpdateParameters"
```

Comment:
>/definitions/ServicesCreateUpdateParameters" [](http://example.com/codeflow?start=24&length=44)

We should use the same schema for both request and response and mark any read-only properties which cannot be part of the request.

Reply:
would fix

Reply:
@j5lim after I made this change, my LintDiff starting showing this error: 

❌ R3006 - BodyTopLevelProperties | Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'DataTransferServiceResourceProperties' has extra properties ['locations'].Location: Microsoft.DocumentDB/preview/2021-04-01-preview/services.json#L318
-- | --
❌ R3006 - BodyTopLevelProperties | Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity, zones. Model definition 'SqlDedicatedGatewayServiceResourceProperties' has extra properties ['sqlDedicatedGatewayEndpoint,locations'].Location: Microsoft.DocumentDB/preview/2021-04-01-preview/services.json#L347

Reply:
I think an array object doesnt work well with x-ms-client-flatten

Reply:
Can you set x-ms-client-flatten to false in DataTransferServiceResource and SqlDedicatedGatewayServiceResource and see if that helps?